ksqldb-parser/src/main/java/io/confluent/ksql/parser/DefaultKsqlParser.java

```diff
@@ -47,7 +51,15 @@ public void syntaxError(
         final String message,
         final RecognitionException e
     ) {
-      throw new ParsingException(message, e, line, charPositionInLine);
+      //Checks if the error is a reserved keyword error
+      if (isKeywordError(message, offendingSymbol)) {
+        final String tokenName = ((CommonToken) offendingSymbol).getText();
```

Comment:
since we're doing this exact check down inside `isKeywordError` we should just pull it out here and pass in the text directly